android/src/main/java/com/survicate/react/SurvicateModuleImpl.java

```diff
@@ -67,4 +71,27 @@ public void reset() {
     public void setWorkspaceKey(String workspaceKey) {
         Survicate.setWorkspaceKey(workspaceKey);
     }
+
+    @ReactMethod
+    public void addListener(String eventName) {
+        if (listenerCount == 0) {
+            Survicate.setEventListener(eventListener);
+        }
+
+        listenerCount++;
+    }
```

Comment:
The evenName is unused

Reply:
It's unused because the parameter itself is required (as this method is overridden) but not needed for any logic. We only have one listener with multiple methods. We would need to have one listener instance per eventName (such as onSurveyDisplayed etc.)